### PR TITLE
Makes pulsing the AI wire of a borg offer the user the option to change it and properly notifies the AI when it is cut.

### LINKED ISSUE #28852: Borgs being unlinked doesn't update in status
[Note]: # (Please enter the commit hash and active testmerge numbers from the "Show Server Revision" verb if you can. If you believe the issue to be caused by a testmerge, please report it in its relative PR thread. State what the issue is from a "whats wrong" prospective. Issue reports should clearly allow maintainers to understand whats wrong and how to test/reproduce if that is not obvious. Avoid ambiguity. Start your issue report below both of these lines (or remove them))

[Admins]: # (If you are reporting a bug that occured AFTER you used varedit/admin buttons to alter an object out of normal operating conditions, please verify that you can re-create the bug without the varedit usage/admin buttons before reporting the issue.)

The past several rounds i have played as an AI and have had roundstart borgs. Due to some circumstance or another, that borg would become unlinked. I wouldn't know about this at all until they tell me, as the status screen still shows them as connected.

### PR DESCRIPTION
**Pull request by Supermichael777**
:cl:
balance: The AI swap menu is now given to the person with the multi-tool rather than the Borg. Old behavior remains for all non multi-tool sources
fix: Borgs and shells now notify when un-linked due to the wire being cut.
/:cl:

Fixes: #28852

The change to the robotics behavior was because it makes more sense for the dude reprogramming you to be able to PICK.

### CODE REVIEW

**Comment by XDTM** [comment]
Feels odd to snipping a wire you technically don't know the function of and somehow being able to decide the outcome. I think robotics should get some kind of programming tablet/device that can interface with open borgs for this purpose, and leave hacking a half-assed way of doing it.

**Comment by Supermichael777** [comment]
its not on snip its on pulse
snip desynchs completly, this is a multitool pulse

File: code/datums/wires/robot.dm [hunk 1]
```diff
@@ -59,8 +63,12 @@
 	switch(wire)
 		if(WIRE_AI) // Cut the AI wire to reset AI control.
 			if(!mend)
+				if(R.shell)
+					R.undeploy()
+					R.notify_ai(AI_SHELL)
```
Comment (GunHog) [inline code/datums/wires/robot.dm]: Have you tested these? It looks like the code would tell the AI that a new borg/shell has been connected when you are in fact disconnecting it. If you want to include messages for borgs disconnected by cutting this wire, you need to make a new message or modify one to indicate the event correct.

Comment (Supermichael777) [inline code/datums/wires/robot.dm]: let me check

**Comment by JJRcop** [comment]
This totally makes sense, because the multitool can probably emit various signals and intensities, which would switch the master AI.

**Comment by Supermichael777** [comment]
I dont know what i did wrong please help

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 2]
```diff
@@ -848,6 +848,8 @@
 			to_chat(connected_ai, "<br><br><span class='notice'>NOTICE - Cyborg reclassification detected: [oldname] is now designated as [newname].</span><br>")
 		if(AI_SHELL) //New Shell
 			to_chat(connected_ai, "<br><br><span class='notice'>NOTICE - New cyborg shell detected: <a href='?src=\ref[connected_ai];track=[html_encode(name)]'>[name]</a></span><br>")
+		if(DISCONNECT) //New Shell
```
Comment (GunHog) [inline code/modules/mob/living/silicon/robot/robot.dm]: You forgot to change this comment.

**Comment by JJRcop** [comment]
`code/datums/wires/robot.dm:39:error: DISCONNECT: undefined var
code/datums/wires/robot.dm:67:error: DISCONNECT: undefined var
code/modules/mob/living/silicon/robot/robot.dm:851:error: DISCONNECT: undefined var`

**Comment by JJRcop** [comment]
Press [Details](https://travis-ci.org/tgstation/tgstation/builds/247789543?utm_source=github_status&utm_medium=notification) to view information like this in the future.

**Comment by Supermichael777** [comment]
I viewed that but i don't know what I did WRONG.

**Comment by JJRcop** [comment]
You never defined what `DISCONNECT` is. Remember, it's not a string.

**Comment by JJRcop** [comment]
Define it in [code/__DEFINES/robots.dm](https://github.com/tgstation/tgstation/blob/5570134c32705d92769bd84cc3483308edf1ae31/code/__DEFINES/robots.dm)

**Comment by Iamgoofball** [comment]
This is a pretty serious nerf

**Comment by GunHog** [comment]
How is this a nerf? The Robo being able to choose the AI rather than the borg?

**Comment by Anonmare** [comment]
Is this stale?

**Comment by Supermichael777** [comment]
its done is what it is

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**